packages/foam-vscode/src/features/create-from-template.ts

```diff
@@ -263,14 +276,29 @@ async function askUserForFilepathConfirmation(
 
 export async function resolveFoamTemplateVariables(
   templateText: string,
-  extraVariablesToResolve: Set<string> = new Set()
+  extraVariablesToResolve: Set<string> = new Set(),
+  givenValues: Map<string, string> = new Map()
 ): Promise<[Map<string, string>, string]> {
-  const givenValues = new Map<string, string>();
   const variables = findFoamVariables(templateText.toString()).concat(
     ...extraVariablesToResolve
   );
   const uniqVariables = [...new Set(variables)];
 
+  const missingFoamVariables = [...givenValues.keys()].filter(
+    x => uniqVariables.indexOf(x) < 0
+  );
+  if (missingFoamVariables.includes('FOAM_SELECTED_TEXT')) {
+    missingFoamVariables.forEach(key => {
```

Comment:
would this add `FOAM_SELECTED_TEXT` once per missing variable? So if there are 3 missing variables we would end up with the selection 3 times in the new note?

Reply:
Ow darn.. sorry 'bout that. Will change it!